src/Microsoft.TestPlatform.Common/Utilities/PEReaderHelper.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.Utilities
+{
+    using System;
+    using System.IO;
+    using System.Reflection.PortableExecutable;
+
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    public class PEReaderHelper
+    {
+        private static PEReaderHelper instance;
+
+        /// <summary>
+        /// Gets the PEReaderHelper instance.
+        /// </summary>
+        internal static PEReaderHelper Instance => instance ?? (instance = new PEReaderHelper());
+
+        /// <summary>
+        /// Determines assembly type from file.
+        /// </summary>
+        public AssemblyType GetAssemblyType(string filePath)
+        {
+            var assemblyType = AssemblyType.None;
+
+            try
+            {
+                using (var assemblyStream = File.Open(filePath, FileMode.Open, FileAccess.Read))
```

Comment:
Use FileHelper interface.

Reply:
Use [FileShare](https://msdn.microsoft.com/en-us/library/y973b725(v=vs.110).aspx).

Reply:
done. Added API in FileHelper also.